kyaml/yaml/types.go

```diff
@@ -35,18 +35,18 @@ func IsMissingOrNull(node *RNode) bool {
 }
 
 // IsEmpty returns true if the RNode is MissingOrNull, or is either a MappingNode with
-// no fields, or a SequenceNode with no elements.
+// no fields.
 func IsEmpty(node *RNode) bool {
-	if node == nil || node.YNode() == nil || node.YNode().Tag == NullNodeTag {
+	if IsMissingOrNull(node) {
```

Comment:
Thanks for the high level krusty test.

Please also add tests to types_test for `IsEmpty` and `IsMissingOrNull`.

We need regression coverage here please; it's foundational stuff and we've had many bugs rooted in it.

Might be useful to structure the tests as
- here's a node with missing or empty fields
- show that the `Is` functions work as expected
- now make that node non-empty / non-missing
- show that the `Is` functions work as expected
- make it empty/ missing again
- show that the `Is` functions work as expected

Can we cover transitions for a list field that is missing, present but empty, present with one entry, present and empty again, then completely deleted? 

Want to cover `NullNodeTag`, empty `MappingNode` and empty `Content` cases.